src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1475,3 +1559,351 @@ function writeManifestFile(
+
+      logInfo(
+        `IPDT patcher ${version} downloaded successfully`,
+        LogPrefix.HyperPlay
+      )
+      await writeFile(versionFile, version)
+
+      if (!isWindows) {
+        await chmod(ipdtPatcher, 0o755)
+      }
+    } catch (error) {
+      captureException(error, {
+        extra: {
+          method: 'downloadPatcher'
+        }
+      })
+
+      logError(`Error downloading IPDT: ${error}`, LogPrefix.HyperPlay)
+      throw new Error('Error downloading IPDT')
+    }
+  }
+}
+
+const getIpdtPatcherVersion = async () => {
+  const { stdout } = await spawnAsync(ipdtPatcher, ['-version'])
+  return `${stdout}`.split(' ')[2]
+}
+
+export async function downloadGameIpdtManifest(
+  appName: string,
+  version: string
+) {
+  try {
+    const {
+      install: { channelName, platform },
+      is_installed
+    } = getGameInfo(appName)
+    if (!channelName || !platform || !is_installed) throw Error('Invalid game')
+
+    // download only if the manifest file is not already downloaded and its the same version
+    const manifestName = `${appName}-${platform}-${version}.json`
+    const manifestPath = path.join(ipdtManifestsPath, manifestName)
+    if (existsSync(manifestPath)) return
+
+    const releaseId = generateID(appName, version)
+    const manifestUrl = await getHyperPlayReleaseManifest(releaseId, platform)
+    if (!manifestUrl) return logWarning(`Manifest not found for ${appName}`)
+
+    // download and save the manifest file as a json file in manifest folder
+    logInfo(
+      `Downloading manifest for ${appName} from ${manifestUrl} to ${manifestPath}`,
+      LogPrefix.HyperPlay
+    )
+    const response = await fetch(manifestUrl)
+    if (!response.ok) {
+      throw new Error(`Failed to download manifest: ${response.statusText}`)
+    }
+    const manifestData = await response.text()
+    await writeFile(manifestPath, manifestData)
+  } catch (error) {
+    logError(
+      `Error downloading manifest for ${appName}: ${error}`,
+      LogPrefix.HyperPlay
+    )
+    captureException(
+      new Error(`Error downloading manifest for ${appName}: ${error}`),
+      {
+        extra: {
+          method: 'downloadGameIpdtManifest',
+          appName,
+          version
+        }
+      }
+    )
+    throw new Error(`Error downloading manifest for ${appName}: ${error}`)
+  }
+}
+
+async function applyPatching(
+  gameInfo: GameInfo,
+  newVersion: string,
+  signal: AbortSignal
+): Promise<InstallResult> {
+  const patcherisEnabled = getFlag('enable-patcher', false)
+
+  if (!patcherisEnabled) {
+    return { status: 'error' }
+  }
+
+  const {
+    app_name: appName,
+    install: { install_path, version, platform }
+  } = gameInfo
+
+  try {
+    const { patchFolder } = await import('@hyperplay/patcher')
+
+    const mainWindow = getMainWindow()
+    let aborted = false
+
+    if (!existsSync(ipdtPatcher)) {
+      await downloadPatcher()
+    }
```

Comment:
I'd say let's download every time to get the latest version while this is feature flagged and open a separate PR to only download if there is a new version 

```suggestion
    await downloadPatcher()
```

Reply:
we'd have to remove this line in `downloadPatcher` too

```
if (!existsSync(ipdtPatcher)) {
```

Reply:
ok, sounds good